tests/Tests/100__deploy/100__installation/102__post_install.robot

```diff
@@ -68,37 +69,45 @@ Verify GPU Operator Deployment  # robocop: disable
     # ...   nvidia-operator-validator DS
 
 Verify That Prometheus Image Is A CPaaS Built Image
-    [Tags]    Sanity   
+    [Tags]    Sanity
     ...     Tier1
-    ...     ODS-734    
+    ...     ODS-734
     ${pod} =    Search Pod    namespace=redhat-ods-monitoring    pod_start_with=prometheus-
     Verify Container Image    redhat-ods-monitoring    ${pod}    prometheus
     ...    "registry.redhat.io/openshift4/ose-prometheus"
     Verify Container Image    redhat-ods-monitoring    ${pod}    oauth-proxy
     ...    "registry.redhat.io/openshift4/ose-oauth-proxy:v4.8"
 
 Verify That Grafana Image Is A Red Hat Built Image
-    [Tags]    Sanity    
+    [Tags]    Sanity
     ...     Tier1
-    ...     ODS-736    
+    ...     ODS-736
     ${pod} =    Search Pod    namespace=redhat-ods-monitoring    pod_start_with=grafana-
     Verify Container Image    redhat-ods-monitoring    ${pod}    grafana
     ...    "registry.redhat.io/rhel8/grafana:7"
     Verify Container Image    redhat-ods-monitoring    ${pod}    auth-proxy
     ...    "registry.redhat.io/openshift4/ose-oauth-proxy:v4.8"
 
 Verify That Blackbox-exporter Image Is A CPaaS Built Image
-    [Tags]    Sanity    
+    [Tags]    Sanity
     ...     Tier1
-    ...     ODS-735    
+    ...     ODS-735
     ${pod} =    Search Pod    namespace=redhat-ods-monitoring    pod_start_with=blackbox-exporter-
     Verify Container Image    redhat-ods-monitoring    ${pod}    blackbox-exporter
     ...    "quay.io/integreatly/prometheus-blackbox-exporter:v0.19.0"
 
 Verify That Alert Manager Image Is A CPaaS Built Image
-    [Tags]    Sanity    
+    [Tags]    Sanity
     ...     Tier1
-    ...     ODS-733    
+    ...     ODS-733
     ${pod} =    Search Pod    namespace=redhat-ods-monitoring    pod_start_with=prometheus-
     Verify Container Image    redhat-ods-monitoring    ${pod}    alertmanager
     ...    "registry.redhat.io/openshift4/ose-prometheus-alertmanager"
+
+Verify Versioning Each Release
+    [Documentation]    Verify versioning each release
+    [Tags]    Sanity
+    ...       ODS-478
+    @{list} =    Oc Get    kind=ClusterServiceVersion    label_selector=olm.copiedFrom=redhat-ods-operator
+    &{dict} =    Set Variable    ${list}[0]
+    Should Match Regexp    ${dict.spec.version}    ^[0-9]+\.[0-9]+\.[0-9]+\-[0-9]+$
```

Comment:
Mh, if I'm correct you're checking for a match for `x.y.z-build`, but I'm not sure the `-build` will always be present. Perhaps that last bit should be optional? Like `^[0-9]+\.[0-9]+\.[0-9]+\-*[0-9]*$`

Reply:
No. the description says we need to check the complete version i.e of the form a.b.c-d

Reply:
The description of what? we are the ones defining the test cases :)
My worry is that this will fail in the case in which there have been no rebuilds of a release, and the version would be x.y.z-0; in that case I *think* that the -0 would be dropped (as it is not communicating anything), but we have never encountered this situation.
In any case, `^[0-9]+\.[0-9]+\.[0-9]+\-*[0-9]*$` would match both `x.y.z-w` and `x.y.z`

Reply:
https://issues.redhat.com/browse/RHODS-1695

Reply:
Yes, the story mentions that the build number increases `for every new build that arrives with bug fixes. `. There might be a release where we don't ever need to rebuild to include bug fixes, and that would mean that its version would be `x.y.z-0`; I expect that in that case `-0` would not be included in the actual build number, as it does not convey any information. With the regex you have currently defined, this would fail the TC.

Reply:
Updated.